master/buildbot/test/unit/test_reporters_pushover.py

```diff
@@ -0,0 +1,386 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import copy
+import sys
+
+from mock import Mock
+
+from twisted.internet import defer
+from twisted.trial import unittest
+
+from buildbot import config
+from buildbot.process.results import CANCELLED
+from buildbot.process.results import EXCEPTION
+from buildbot.process.results import FAILURE
+from buildbot.process.results import SUCCESS
+from buildbot.process.results import WARNINGS
+from buildbot.reporters import utils
+from buildbot.reporters.pushover import PushoverNotifier
+from buildbot.test.fake import httpclientservice as fakehttpclientservice
+from buildbot.test.fake import fakedb
+from buildbot.test.fake import fakemaster
+from buildbot.test.util.config import ConfigErrorsMixin
+
+py_27 = sys.version_info[0] > 2 or (sys.version_info[0] == 2
+                                    and sys.version_info[1] >= 7)
+
+
+class TestPushoverNotifier(ConfigErrorsMixin, unittest.TestCase):
+
+    def setUp(self):
+        self.master = fakemaster.make_master(testcase=self,
+                                             wantData=True, wantDb=True, wantMq=True)
+        self._http = self.successResultOf(fakehttpclientservice.HTTPClientService.getFakeService(
+            self.master, self, 'https://api.pushover.net'))
+
+    @defer.inlineCallbacks
+    def setupBuildResults(self, results, wantPreviousBuild=False):
+        # this testsuite always goes through setupBuildResults so that
+        # the data is sure to be the real data schema known coming from data
+        # api
+
+        self.db = self.master.db
+        self.db.insertTestData([
+            fakedb.Master(id=92),
```

Comment:
Sorry I did not see this test, as it was hidden by default by github.

No need to retest the behaviour for pushover. It should also be factorized in a test for notifier.py

In unit/test_reporter_pushover, you should only test the correctness of send message, so you dont need to create all the fake data in fakedb.